src/upchunk.spec.ts

```diff
@@ -1,333 +1,322 @@
+  it('a single chunk failing the max number of times fails the upload', (done) => {
+    xhrMock.put(
+      endpoint,
+      sequence([
+        { status: 502 },
+        { status: 502 },
+        { status: 502 },
+        { status: 502 },
+        { status: 502 },
+        { status: 200 },
+      ])
+    );
 
-  const successCallback = jest.fn();
+    const upload = createUploadFixture({ delayBeforeAttempt: 0.1 });
 
-  upload.on('chunkSuccess', successCallback);
+    upload.on('error', (err) => {
+      try {
+        expect(err.detail.chunk).to.equal(0);
+        expect(err.detail.attempts).to.equal(5);
+        done();
+      } catch (err) {
+        done(err);
+      }
+    });
 
-  upload.on('success', () => {
-    expect(successCallback).toBeCalledTimes(2);
-    done();
+    upload.on('success', () => {
+      done(new Error(`Expected upload to fail due to failed attempts`));
+    });
   });
-});
 
-const isNumberArraySorted = (a: number[]): boolean => {
-  for (let i = 0; i < a.length - 1; i += 1) {
-    if (a[i] > a[i + 1]) {
-      return false;
-    }
-  }
-  return true;
-};
-
-test('progress event fires the correct upload percentage', (done) => {
-  const fileBytes = 1048576;
-  const upload = createUploadFixture(
-    {},
-    new File([new ArrayBuffer(fileBytes)], 'test.mp4')
-  );
-
-  const scopes = [
-    nock('https://example.com')
-      .matchHeader('content-range', `bytes 0-${fileBytes / 4 - 1}/${fileBytes}`)
-      .put('/upload/endpoint')
-      .reply(200),
-    nock('https://example.com')
-      .matchHeader(
-        'content-range',
-        `bytes ${fileBytes / 4}-${fileBytes / 2 - 1}/${fileBytes}`
-      )
-      .put('/upload/endpoint')
-      .reply(200),
-    nock('https://example.com')
-      .matchHeader(
-        'content-range',
-        `bytes ${fileBytes / 2}-${3 * fileBytes / 4 - 1}/${fileBytes}`
-      )
-      .put('/upload/endpoint')
-      .reply(200),
-    nock('https://example.com')
-      .matchHeader(
-        'content-range',
-        `bytes ${3 * fileBytes / 4}-${fileBytes - 1}/${fileBytes}`
-      )
-      .put('/upload/endpoint')
-      .reply(200),
-  ];
-
-  const progressCallback = jest.fn((percentage) => percentage);
-
-  upload.on('error', (err) => {
-    done(err);
-  });
+  it('chunkSuccess event is fired after each successful upload', (done) => {
+    let calls = 0;
+    xhrMock.put(endpoint, { status: 200 });
 
-  upload.on('progress', (progress) => {
-    progressCallback(progress.detail);
+    const upload = createUploadFixture();
+
+    upload.on('chunkSuccess', () => {
+      calls++;
+    });
+
+    upload.on('success', () => {
+      expect(calls).to.equal(2);
+      done();
+    });
   });
 
-  upload.on('success', () => {
-    scopes.forEach((scope) => {
-      if (!scope.isDone()) {
-        done('All scopes not completed');
+  /** @TODO Figure out how to best handle progress testing. See: https://www.npmjs.com/package/xhr-mock#upload-progress (CJP) */
```

Comment:
unfortunately, xhr-mock's progress events may not be fully-fleshed out enough for our needs here. https://github.com/jameslnewell/xhr-mock/issues/82

An alternative is https://github.com/berniegp/mock-xmlhttprequest/#simulating-progress